server/repositories/__tests__/prisonApi.spec.js

```diff
@@ -0,0 +1,221 @@
+      }
+
+      expect(hasThrown).toBe(true);
+    });
+
+    it('should swallow the error and return nothing when the request fails', async () => {
+      const repository = new PrisonApiRepository({ client, apiUrl });
+
+      client.get.mockImplementation(() => Promise.rejects('💥'));
+
+      const response = await repository.getTransactionsFor(
+        'A123BC',
+        'spends',
+        new Date('2020-01-01'),
+        new Date('2020-01-31'),
+      );
+
+      expect(response).toBeNull();
+    });
+  });
+
+  describe('getPrisonDetailsFor', () => {
+    it('should return when the prison details request succeeds', async () => {
+      const repository = new PrisonApiRepository({ client, apiUrl });
+
+      client.get.mockImplementation(() => Promise.resolve('API_RESPONSE'));
+
+      const response = await repository.getPrisonDetailsFor('TST');
+
+      expect(lastCall(client.get)[0]).toContain('/agencies/TST');
+
+      expect(response).toBe('API_RESPONSE');
+    });
+
+    it('should throw when no prisonId is passed', async () => {
+      const repository = new PrisonApiRepository({ client, apiUrl });
+
+      client.get.mockImplementation(() => Promise.resolve('API_RESPONSE'));
+
+      let hasThrown = false;
+
+      try {
+        await repository.getPrisonDetailsFor();
+      } catch (e) {
+        hasThrown = true;
+        expect(client.get).not.toHaveBeenCalled();
+      }
+
+      expect(hasThrown).toBe(true);
+    });
+
+    it('should swallow the error and return nothing when the request fails', async () => {
+      const repository = new PrisonApiRepository({ client, apiUrl });
+
+      client.get.mockImplementation(() => Promise.rejects('💥'));
+
+      const response = await repository.getPrisonDetailsFor('TST');
+
+      expect(response).toBeNull();
+    });
+  });
+
+  describe('getBalancesFor', () => {
+    it('should return when the balances request succeeds', async () => {
+      const repository = new PrisonApiRepository({ client, apiUrl });
+
+      client.get.mockImplementation(() => Promise.resolve('API_RESPONSE'));
+
+      const response = await repository.getBalancesFor('123456');
+
+      expect(lastCall(client.get)[0]).toContain('/bookings/123456/balances');
+
+      expect(response).toBe('API_RESPONSE');
+    });
+
+    it('should throw when no bookingId is passed', async () => {
+      const repository = new PrisonApiRepository({ client, apiUrl });
+
+      client.get.mockImplementation(() => Promise.resolve('API_RESPONSE'));
+
+      let hasThrown = false;
+
+      try {
+        await repository.getBalancesFor();
+      } catch (e) {
+        hasThrown = true;
+        expect(client.get).not.toHaveBeenCalled();
+      }
+
+      expect(hasThrown).toBe(true);
+    });
+
+    it('should swallow the error and return nothing when the request fails', async () => {
+      const repository = new PrisonApiRepository({ client, apiUrl });
+
+      client.get.mockImplementation(() => Promise.rejects('💥'));
```

Comment:
and similarly: `client.get.mockRejectedValue`

Reply:
Totally agree here, prefer to use the Jest in-builts when possible. Updated the tests